Autopilot/Inc/ultrasonic.hpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * Ultrasonic Sensor Functions
+ * Author: Dhruv Upadhyay, February 2021
+ */
+
+ #ifndef ULTRASONIC_HPP
+ #define ULTRASONIC_HPP
+
+ #include <stdint.h>
+
+typedef struct {
+    float distance;
+    bool isDataNew; // Holds whether the sensor data has been updated or not
+    uint8_t status; // -1 for trig not sent and echo not received, 0 for trig sent, echo not received, and 1 for echo received
+    float time; // represents the time taken between trig being sent and echo being received
```

Comment:
Is there a reason both time and distance exist as well?

Reply:
Time is an intermediate variable used to find the distance, tbh I should probably remove it from the struct as it's not a useful part of the final data